deployment/terraform/cloudfront-functions/conditional_oshub_redirect.js

```diff
@@ -0,0 +1,71 @@
+// Adapted from https://devscope.io/code/aws-samples/amazon-cloudfront-functions/issues/11
+
+function objectToQueryString(obj) {
+    var str = [];
+    for (var param in obj)
+        if (obj[param].value === '') {
+            str.push(encodeURIComponent(param));
+        } else {
+            // Normalize single and multiple value query params to a list of
+            // param value objects
+            var valueObjects = obj[param].multiValue || [obj[param]];
+            var encodedValues = valueObjects.map(
+                vo => `${encodeURIComponent(param)}=${encodeURIComponent(vo.value)}`);
+
+            str.push(encodedValues.join('&'));
+        }
+
+    return str.join('&');
+}
+
+function handler(event) {
+    var request = event.request;
+    var uri = request.uri;
+
+    // If this is an embed request, allow it to proceed to the ALB
+    if (request.querystring.embed && request.querystring.embed.value === '1') {
+        return request;
+    }
+
+
+    // Ensure that any resources fetched by the client after when loading an
+    // ebedded map are not redirected
+    if (uri.substring(1, 5) === 'web/'
+        || uri.substring(1, 5) === 'api/'
+        || uri.substring(1, 19) === 'api-feature-flags/'
+        || uri.substring(1, 10) === 'facvicon/'
```

Comment:
Should this be favicon?

Reply:
Also, what is the best way to test these? It looks like `/api/facilities/` is expected to redirect when it's the URL path, so I assume testing this is a different procedure?

Reply:
The two nested `if` checks workin combination. The paths in the first `if` block are resources that are requested by the embedded map after it loads. If one browses `https://openapparel.org/api/*` directly the `referer` header is not set, and we will redirect them to OSH. When the embedded map page makes requests to `/api/*` to load dropdowns, etc. the `referer` header is set, so we let the request pass through.

Reply:
favicon typo fixed up and rebased. Thank you.